pkg/odo/cli/cli.go

```diff
@@ -171,14 +177,56 @@ func NewCmdOdo(name, fullName string) *cobra.Command {
 		config.NewCmdConfiguration(config.RecommendedCommandName, util.GetFullName(fullName, config.RecommendedCommandName)),
 		preference.NewCmdPreference(preference.RecommendedCommandName, util.GetFullName(fullName, preference.RecommendedCommandName)),
 		debug.NewCmdDebug(debug.RecommendedCommandName, util.GetFullName(fullName, debug.RecommendedCommandName)),
-		component.NewCmdPushDevfile(component.PushDevfileRecommendedCommandName, util.GetFullName(fullName, component.PushDevfileRecommendedCommandName)),
 	)
 
+	// Expose commands in experimental mode, if experimental mode is enabled.
+	rootCmd = addExperimentalCommands(rootCmd, fullName)
+
 	odoutil.VisitCommands(rootCmd, reconfigureCmdWithSubcmd)
 
 	return rootCmd
 }
 
+// addExperimentalCommands exposes experimental commands which are in developement or experimental mode,
+// if experimental mode has been set via odo preferences or env variable
+// by default experimental mode is disabled
+func addExperimentalCommands(rootCmd *cobra.Command, fullName string) *cobra.Command {
```

Comment:
`rootCmd` does not make sense as a name. Is it not possible to pass any command technically to this function. And if the function is specifically meant to add experimental commands to only the rootCmd, as it is here, it might make sense if the function name makes it explicitly clear of the intent. I could read the name and mistake it for a function that generally adds experimental subcommands to any command :)

Reply:
@mohammedzee1000 , I think you have a valid point, it should be possible to pass any command to this function. Let me see, how can we accommodate that. 

This also opens up the possibility for adding subcommands into experimental mode. This would require some refactor though.

Thanks for the comment :)

Reply:
I have changed the design for the experimental feature, the new design is much simplified and easy to use. It can be leveraged for any sub command as well.